### ISSUE TITLE
Feature: foreground and background colour combinations provide sufficient contrast 

### ISSUE THREAD

**Issue by pranshuchittora** [react-native-elements/react-native-elements#2746]
**Is your feature request related to a problem? Please Describe.**
Feature: Ensure that foreground and background colour combinations provide sufficient contrast when viewed by someone having colour deficits or when viewed on a black and white screen

Currently, the developer has to keep the colour of the text and background in mind to ensure good contrast and differentiability of the content from the background.

https://www.w3.org/TR/AERT/#color-contrast

**Describe the solution you'd like**
A prop for components which are user interactable like `smartColor` or something like that which enables such modifications.

**Comment by pranshuchittora** [comment]
Yep, so for the provided backgroundColor, we need to generate or choose from white or black for the colour of the Text

**Comment by flyingcircle** [comment]
This more or less seems like it is already provided by the `color` library: https://www.npmjs.com/package/color People can auto generate different colors based on their own preferences. Unless we can solve a specific problem with this, this seems to already be solved in a very common library.
